soda_machine.rb

```diff
@@ -6,13 +8,27 @@ def initialize(args = {})
     @cash = args[:cash]
   end
 
+  # return the number of sodas it contains
   def current_inventory_count
+    @sodas.length
   end
 
+  # find single soda with a given brand
   def find_soda(soda_brand)
+    @sodas.find {|soda| soda.brand == soda_brand}
   end
 
+  # sell a soda with a given brand
   def sell(soda_brand)
+    return nil unless
+      @sold_soda = find_soda(soda_brand)
```

Comment:
couple of points:
- no need to put these on two separate lines, could be on the same line like this
- also no need for an instance variable here since `sold_soda` is only used locally.
- finally, i would add some parens to clear up the order of operations.

```ruby
return nil unless (sold_soda = find_soda(soda_brand))
```

even better would be to do this on two lines like this:

```ruby
sold_soda = find_soda(soda_brand)
return nil unless sold_soda
```